### ISSUE TITLE
[QUESTION]The ReadOptions uses the different snapshots when getting metadata and subkey values

### ISSUE THREAD

**Issue by ColinChamber** [apache/kvrocks#519]
The ReadOptions uses the different snapshots when getting metadata and subkey values. Could there be inconsistencies?

**Comment by git-hulk** [comment]
Yes, but it seems ok for most scenes, implementation in this way can keep the code more clean and simple.

**Comment by ShooterIT** [comment]
I think we should avoid this, it is a bad design. We should find a time to refactor this implementation.
